### PR TITLE
VideoPress: Add transform control from video block to videopress/video block

### LINKED ISSUE #28637: VideoPress: Include a manual conversion panel to/from v5/v6
Let's remove the auto-conversion process of v5/v6, and add the ability to convert the block manually adding a new panel in the block sidebar. Let's do this via filters.

### PR DESCRIPTION
**Pull request by dhasilva**
Fixes #28637

## Proposed changes:

* Temporarily adds a transform control to v5 video blocks containing VideoPress-converted videos to v6 to encourage update

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1675244289505449/1675182827.670519-slack-C03TA48NSUX

## Testing instructions:

* Before connecting Jetpack, go to the block editor, add a `video` block and upload a video. Save the post.
* Connect Jetpack.
* Add two `video` block, uploading two videos to be processed by VideoPress.
* Add a v6 `videopress/video` block with another video.
* Save the post.
* Check that only the `video` blocks with a VideoPress-converted video display the transform control.
* In the transform control of one of the `video` blocks, click on `Transform blocks to VideoPress`.
* Check that the two `video` blocks processed by VideoPress were transformed to v6 `videopress/video` blocks without loss of data.

https://user-images.githubusercontent.com/8486249/216062108-076b1f1f-1a4a-447f-bb54-fd167bda1e0d.mp4

### CODE REVIEW

**Review by retrofox — COMMENTED** [review]
We need to check if the video block is also `v5`. You can use [isVideoPressBlockBasedOnAttributes()](https://github.com/Automattic/jetpack/blob/3280832de60155e74990d2b762a135706dab0a21/projects/plugins/jetpack/extensions/blocks/videopress/editor.js#L486).

File: projects/plugins/jetpack/extensions/blocks/videopress/v6-transform/components/transform-control/index.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	const { getBlocks } = useSelect( blockEditorStore );
+	const { replaceBlock } = useDispatch( blockEditorStore );
+
+	const handleTransformAll = () => {
+		const blocks = getBlocks();
+
+		blocks.forEach( block => {
+			const { clientId, name, attributes } = block;
+
+			if ( name === 'core/video' && attributes.guid ) {
```
Comment (retrofox) [inline projects/plugins/jetpack/extensions/blocks/videopress/v6-transform/components/transform-control/index.js]: Should we use also here the `isVideoPressBlockBasedOnAttributes()` helper?

Comment (dhasilva) [reply projects/plugins/jetpack/extensions/blocks/videopress/v6-transform/components/transform-control/index.js]: Good catch!

File: projects/plugins/jetpack/extensions/blocks/videopress/editor.js [hunk 2]
```diff
@@ -547,16 +531,18 @@ const convertVideoBlockToVideoPressVideoBlock = createHigherOrderComponent( Bloc
...
 		const shouldConvertCoreVideoToVideoPressVideoBlock = !! (
 			isCoreVideoBlock && // Only auto-convert if the block is a core/video block
 			isVideoPressVideoBlockRegistered && // Only auto-convert if the VideoPress block is registered
 			isCoreVideoVideoPressBlock && // Only auto-convert if the block is a VideoPress block
 			isVideoPressVideoBlockAvailable && // Only auto-convert if the feature is available
-			// Only auto-convert if the site is Simple
-			isSimple
+			isSimple && // Only auto-convert if the site is Simple
+			// Disable auto-conversion for now.
+			false
```
Comment (retrofox) [inline projects/plugins/jetpack/extensions/blocks/videopress/editor.js:542]: 💯

**Review by retrofox — APPROVED** [review]
Tested in Jetpack, Atomic, and Simple sites. LGTM. 🚢

**Pull request merged**